Feature/like

SeoYeonLee12 opened the pull request:
도서 좋아요 api입니다. 
도서 좋아요 생성, 생성된 좋아요 삭제, 도서 좋아요 수 조회 기능을 추가했습니다.

seung-gyu-kim committed:
feat: Category Entity에 depth 추가

seung-gyu-kim committed:
refactor: JUnit5 test classes and methods should have default package visibility

test 코드에서 접근제한자 default 사용

seung-gyu-kim committed:
feat: 도서 카데고리 등록 api 구현

Resolves: #50

seung-gyu-kim committed:
feat: 도서 카데고리 등록 api 구현

Resolves: #50

seung-gyu-kim committed:
feat: 도서 카데고리 등록 api 구현

Resolves: #50

seung-gyu-kim committed:
feat: 도서 카테고리 등록 api 구현

Resolves: #50

seung-gyu-kim committed:
feat: 도서 카테고리 등록 api 구현

Resolves: #50

seung-gyu-kim committed:
feat: 도서 카데고리 등록 api 구현

Resolves: #50

seung-gyu-kim committed:
feat: 도서 카테고리 등록 api 구현

Resolves: #50

seung-gyu-kim committed:
feat: validation 추가

Resolves: #50

seung-gyu-kim committed:
feat: validation 추가

Resolves: #50

seung-gyu-kim committed:
feat: validation 추가

Resolves: #50

seung-gyu-kim committed:
feat: validation 추가

Resolves: #50

seung-gyu-kim committed:
feat: validation 추가

Resolves: #50

seung-gyu-kim committed:
feat: validation 추가

Resolves: #50

seung-gyu-kim committed:
feat: validation 추가

Resolves: #50

seung-gyu-kim committed:
feat: validation 추가

Resolves: #50

seung-gyu-kim committed:
feat: validation 추가

Resolves: #50

seung-gyu-kim committed:
feat: validation 추가

Resolves: #50

seung-gyu-kim committed:
feat: validation 추가

Resolves: #50

seung-gyu-kim committed:
feat: validation 추가

Resolves: #50

seung-gyu-kim committed:
feat: validation 추가

Resolves: #50

seung-gyu-kim committed:
feat: validation 추가

Resolves: #50

seung-gyu-kim committed:
feat: validation 추가

Resolves: #50

seung-gyu-kim committed:
feat: validation 추가

Resolves: #50

seung-gyu-kim committed:
feat: category dto에 depth 추가

seung-gyu-kim committed:
feat: category dto에 depth 추가

seung-gyu-kim committed:
feat: category dto에 depth 추가

seung-gyu-kim committed:
feat: 도서 카데고리 등록 api 구현

Resolves: #50

seung-gyu-kim committed:
feat: 공통 응답처리 구현

seung-gyu-kim committed:
feat: 공통 응답처리 구현

seung-gyu-kim committed:
feat: 공통 응답처리 구현

seung-gyu-kim committed:
feat: 권한 검사 구현

Resolves: #50

seung-gyu-kim committed:
merge test 충돌 해결

seung-gyu-kim committed:
feat: 카테고리 조회 api 구현 중

공통화 처리를 위한 진행과정 커밋

SeoYeonLee12 committed:
{feature}: Like 기능 생성- repository, service layer

SeoYeonLee12 committed:
{feature}: Like 기능 생성- repository, service layer

SeoYeonLee12 committed:
{feature}: Like 기능 생성- repository, service layer

SeoYeonLee12 committed:
{feature}: Like 기능 생성- repository, service layer

SeoYeonLee12 committed:
{feature}: Like 기능 생성- repository, service layer

SeoYeonLee12 committed:
{feature}: Like 기능 생성- repository, service layer

SeoYeonLee12 committed:
{feature}: Like 기능 생성- repository, service layer

SeoYeonLee12 committed:
{feature}: Like 기능 생성- repository, service layer

SeoYeonLee12 committed:
{feature}: Like 기능 생성- repository, service layer

SeoYeonLee12 committed:
{feature}: Like 기능 생성- repository, service layer

SeoYeonLee12 committed:
{feature}: Like 기능 생성- repository, service layer

SeoYeonLee12 committed:
Merge remote-tracking branch 'origin/develop' into feature/like

seung-gyu-kim committed:
feat: 관리자 권한 검사 인터셉

seung-gyu-kim committed:
test: 관리자 권한 검사 테스트 수정

seung-gyu-kim committed:
feat: open, admin interceptor 개발

SeoYeonLee12 committed:
{feature}:controller, Test 설정

SeoYeonLee12 committed:
{feature}:controller, Test 설정

SeoYeonLee12 committed:
{feature}:controller, Test 설정

SeoYeonLee12 committed:
{feature}:controller, Test 설정

SeoYeonLee12 committed:
{feature}:controller, Test 설정

SeoYeonLee12 committed:
{feature}:controller, Test 설정

seung-gyu-kim committed:
feat: 카테고리 관리자 화면에서 조회용 api 개발

seung-gyu-kim committed:
fix: ResponseBodyAdvice 수정

ResponseEntity.ok 로 감쌀경우 추가 정보가 들어가서 feingClient Decoding error 발생

seung-gyu-kim committed:
fix: ResponseBodyAdvice 수정

ResponseEntity.ok 로 감쌀경우 추가 정보가 들어가서 feingClient Decoding error 발생

seung-gyu-kim committed:
fix: ResponseBodyAdvice 수정

ResponseEntity.ok 로 감쌀경우 추가 정보가 들어가서 feingClient Decoding error 발생

seung-gyu-kim committed:
fix: SLF4J: Class path contains multiple SLF4J bindings. 에러 해결

slf4j-log4j12 종속성 제거

seung-gyu-kim committed:
feat: 관리자 카테고리 조회 기능 개발

seung-gyu-kim committed:
feat: 관리자 카테고리 조회 기능 개발

seung-gyu-kim committed:
feat: 관리자 카테고리 조회 기능 개발

seung-gyu-kim committed:
feat: 관리자 카테고리 조회 기능 개발

seung-gyu-kim committed:
feat: 관리자 카테고리 조회 기능 개발

seung-gyu-kim committed:
feat: 관리자 카테고리 조회 기능 개발

seung-gyu-kim committed:
feat: 관리자 카테고리 조회 기능 개발

seung-gyu-kim committed:
feat: 관리자 카테고리 조회 기능 개발

seung-gyu-kim committed:
feat: 관리자 카테고리 조회 기능 개발

seung-gyu-kim committed:
feat: 관리자 카테고리 조회 기능 개발

seung-gyu-kim committed:
feat: 관리자 카테고리 조회 기능 개발

seung-gyu-kim committed:
feat: 관리자 카테고리 조회 기능 개발

seung-gyu-kim committed:
feat: 관리자 카테고리 조회 기능 개발

seung-gyu-kim committed:
feat: 관리자 카테고리 조회 기능 개발

seung-gyu-kim committed:
feat: 관리자 카테고리 조회 기능 개발

seung-gyu-kim committed:
feat: 관리자 카테고리 조회 기능 개발

seung-gyu-kim committed:
feat: 관리자 카테고리 추가에서 depth 제거

관리자 카테고리 추가시 depth는 parent id + 1이 되도록 수정

SongAhyeon01 committed:
feat: surefire & jacoco 적용

SeoYeonLee12 committed:
{add}: Like 기능 추가

SeoYeonLee12 committed:
{add}: Like 기능 추가

SeoYeonLee12 committed:
{add}: Like 기능 추가

SeoYeonLee12 committed:
{add}: Like 기능 추가

SeoYeonLee12 committed:
{add}: Like 기능 추가

SeoYeonLee12 committed:
{add}: Like 기능 추가

SeoYeonLee12 committed:
{add}: Like 기능 추가

SeoYeonLee12 committed:
{add}: Like 기능 추가

SeoYeonLee12 committed:
merge conflict 해결

SeoYeonLee12 committed:
{add}: addLike api 생성

SeoYeonLee12 committed:
{add}: addLike api 생성

SeoYeonLee12 committed:
{add}: addLike api 생성

SeoYeonLee12 committed:
{add}: addLike api 생성

SeoYeonLee12 committed:
{add}: addLike api 생성

SeoYeonLee12 committed:
Merge remote-tracking branch 'origin/feature/like' into feature/like

# Conflicts:
#	src/docs/asciidoc/category/admin-category.adoc
#	src/docs/asciidoc/index.adoc
#	src/main/java/com/nhnacademy/store99/bookstore/category/controller/CategoryAdminController.java
#	src/main/java/com/nhnacademy/store99/bookstore/category/dto/request/ModifyCategoryRequest.java
#	src/main/java/com/nhnacademy/store99/bookstore/category/entity/Category.java
#	src/main/java/com/nhnacademy/store99/bookstore/category/repository/CategoryRepository.java
#	src/main/java/com/nhnacademy/store99/bookstore/category/service/CategoryAdminService.java
#	src/main/java/com/nhnacademy/store99/bookstore/category/service/impl/CategoryAdminServiceImpl.java
#	src/main/java/com/nhnacademy/store99/bookstore/common/advice/CommonRestControllerAdvice.java
#	src/main/java/com/nhnacademy/store99/bookstore/config/WebConfig.java
#	src/main/java/com/nhnacademy/store99/bookstore/like/controller/LikeController.java
#	src/main/java/com/nhnacademy/store99/bookstore/like/service/impl/LikeServiceImpl.java
#	src/test/java/com/nhnacademy/store99/bookstore/category/controller/CategoryAdminControllerTest.java
#	src/test/java/com/nhnacademy/store99/bookstore/category/repository/CategoryRepositoryTest.java
#	src/test/java/com/nhnacademy/store99/bookstore/category/service/impl/CategoryAdminServiceImplTest.java

SeoYeonLee12 committed:
{add}: deleteLike api 생성

SeoYeonLee12 committed:
{add}: deleteLike api 생성

SeoYeonLee12 committed:
{add}: deleteLike api 생성

SeoYeonLee12 committed:
{add}: deleteLike api 생성

SeoYeonLee12 committed:
{add}: deleteLike api 생성

SeoYeonLee12 committed:
{add}: deleteLike api 생성

SeoYeonLee12 committed:
{add}: getCnt api 생성

SeoYeonLee12 committed:
{add}: getCnt api 생성

SeoYeonLee12 committed:
{add}: getCnt api 생성

SeoYeonLee12 committed:
{add}: getLikeCnt api 생성

SeoYeonLee12 committed:
{add}: getLikeCnt api 생성

SeoYeonLee12 committed:
{add}: getLikeCnt api 생성

SeoYeonLee12 committed:
{add}: getLikeCnt api 생성

SeoYeonLee12 committed:
{add}: getLikeCnt api 생성

SeoYeonLee12 committed:
{add}: getLikeCnt api 생성

SeoYeonLee12 committed:
{add}: getLikeCnt api 수정

SeoYeonLee12 committed:
{add}: getLikeCnt api 수정

SeoYeonLee12 committed:
{add}: getLikeCnt api 수정

SeoYeonLee12 committed:
{add}: getLikeCnt api 수정

SeoYeonLee12 committed:
{add}: getLikeCnt api 수정

SeoYeonLee12 committed:
{add}: getLikeCnt api 수정

SeoYeonLee12 committed:
Merge remote-tracking branch 'origin/feature/like' into feature/like

# Conflicts:
#	src/main/java/com/nhnacademy/store99/bookstore/like/controller/LikeController.java
#	src/main/java/com/nhnacademy/store99/bookstore/like/repository/LikeRepository.java
#	src/main/java/com/nhnacademy/store99/bookstore/like/service/LikeService.java
#	src/main/java/com/nhnacademy/store99/bookstore/like/service/impl/LikeServiceImpl.java

SeoYeonLee12 committed:
{add}: BookRepository 추가

SeoYeonLee12 committed:
{add}: BookRepository 추가

SeoYeonLee12 committed:
Delete src/test/java/com/nhnacademy/store99/bookstore/like/service/impl/LikeServiceImplTest.java

SeoYeonLee12 committed:
Delete src/test/java/com/nhnacademy/store99/bookstore/like/service/impl/LikeServiceImplTest.java

SeoYeonLee12 committed:
Delete src/test/java/com/nhnacademy/store99/bookstore/like/service/impl/LikeServiceImplTest.java

SeoYeonLee12 committed:
Delete src/test/java/com/nhnacademy/store99/bookstore/like/service/impl/LikeServiceImplTest.java

SeoYeonLee12 committed:
Delete src/test/java/com/nhnacademy/store99/bookstore/like/repository/LikeRepositoryTest.java

SeoYeonLee12 committed:
Delete src/test/java/com/nhnacademy/store99/bookstore/like/repository/LikeRepositoryTest.java

SeoYeonLee12 committed:
Delete src/test/java/com/nhnacademy/store99/bookstore/like/repository/LikeRepositoryTest.java

SeoYeonLee12 committed:
Delete src/test/java/com/nhnacademy/store99/bookstore/like/repository/LikeRepositoryTest.java

seung-gyu-kim commented on src/main/java/com/nhnacademy/store99/bookstore/like/controller/LikeController.java:
```diff
@@ -0,0 +1,76 @@
+    public ResponseEntity<CommonResponse<Void>> postAddLike(@RequestBody @Valid LikeRequest req) {
+        likeService.addLike(req);
+        CommonHeader commonHeader = CommonHeader.builder()
+                .httpStatus(HttpStatus.CREATED)
+                .resultMessage("Successfully add like")
+                .build();
+        CommonResponse<Void> commonResponse = CommonResponse.<Void>builder()
+                .header(commonHeader)
+                .build();
+        return ResponseEntity.ok(commonResponse);
+
+    }
+
+    @DeleteMapping("/{likeId}")
+    public ResponseEntity<CommonResponse<String>> postDeleteLike(@RequestBody @Valid @PathVariable Long likeId) {
+        String response = likeService.deleteLike(likeId);
+        CommonHeader commonHeader = CommonHeader.builder()
+                .httpStatus(HttpStatus.OK)
+                .resultMessage("Successfully delete like")
+                .build();
+        CommonResponse<String> commonResponse = CommonResponse.<String>builder()
+                .header(commonHeader)
+                .result(response)
+                .build();
+        return ResponseEntity.ok(commonResponse);
+    }
+
+    @GetMapping("/likeCnt")
+    public ResponseEntity<CommonResponse<Long>> getCnt(@RequestBody @Valid Long bookId) {
+        Long cnt = likeService.countLikesByBookId(bookId);
+        CommonHeader commonHeader = CommonHeader.builder()
+                .httpStatus(HttpStatus.OK)
+                .resultMessage("Success")
+                .build();
+        CommonResponse<Long> commonResponse = CommonResponse.<Long>builder()
+                .header(commonHeader)
+                .result(cnt)
+                .build();
+        return ResponseEntity.ok(commonResponse);
+    }
```
```java
CommonHeader commonHeader = CommonHeader.builder()
                .httpStatus(HttpStatus.OK)
                .resultMessage("Success")
                .build();
        CommonResponse<Long> commonResponse = CommonResponse.<Long>builder()
                .header(commonHeader)
                .result(cnt)
                .build();
```
저희 프로젝트에는 매번 반복되는 위와 같은 과정을 ResponseBodyAdvice를 통해 공통화 하였습니다. 다음에는 객체를 바로 반환해보세요!

seung-gyu-kim commented on src/main/java/com/nhnacademy/store99/bookstore/like/dto/request/LikeRequest.java:
```diff
@@ -0,0 +1,14 @@
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+@Data
```
`@Data`는 너무 많은 롬복이 얽혀있습니다. 저희 컨벤션에 무분별한 롬복 사용하지 않기(https://github.com/nhnacademy-be5-staff99#%EA%B3%B5%ED%86%B5) 라는 내용이 있으니 필요한 내용만 선택하는 것이 어떨까요?

seung-gyu-kim commented on src/main/java/com/nhnacademy/store99/bookstore/like/dto/response/LikeResponse.java:
```diff
@@ -0,0 +1,13 @@
+import lombok.Data;
+
+@AllArgsConstructor
+@Data
```
위와 동

seung-gyu-kim commented on src/main/java/com/nhnacademy/store99/bookstore/like/service/impl/LikeServiceImpl.java:
```diff
@@ -0,0 +1,83 @@
+        return likeRepository.countLikesByBookId(bookId);
+    }
+
+    //    @Override
+//    @Transactional
+//    public Page<BookInfoForLikeResponse> getAllByUser(Long userId, Pageable pageable) {
+//        likeRepository.findAllByUserId(userId, pageable).getPageable();
+//        return null;
+//    }
```
차후 개발 예정이신거죠? 안쓰실 예정이라면 제거해주세용

SeoYeonLee12 replied:
넵 조회 Api 개발 예정입니다!

seung-gyu-kim commented on src/main/java/com/nhnacademy/store99/bookstore/like/service/impl/LikeServiceImpl.java:
```diff
@@ -0,0 +1,83 @@
+            throw new IllegalStateException("Noting deleted the like!");
+        }
+        likeRepository.deleteById(likeId);
+        return null;
```
void를 사용하지 않으시고 String으로 하여 null을 반환할 이유가 따로 있나요?

SeoYeonLee12 replied:
수정했어야 했는데 못했습니다! 수정했어요!

seung-gyu-kim reviewed (approved):
아직 테스트 코드가 없는 것 같습니다! 혹시 빼먹고 올리신걸까요? 아니면 다음에 따로 올리실 예정이신건가요?

SongAhyeon01 committed:
feat: 메인 마이페이지 api

- 마이페이지 접근 시 기본적으로 이름, 이메일, 연락처, 생년월일, 주소, 현재등급 및 등급 정보, 현재 보유 포인트를 반환

- X-USER-ID header 로 로그인 한 사용자를 파악하고, 해당 유저의 정보를 조회

#89

SeoYeonLee12 committed:
Merge remote-tracking branch 'origin/feature/like' into feature/like

SeoYeonLee12 committed:
{style}: code reformat

SeoYeonLee12 committed:
conflict 해결

SeoYeonLee12 committed:
{refector}: code review 적용

SeoYeonLee12 committed:
Merge branch 'develop' into feature/like

Rosin23 reviewed (approved):
좋아요 기능이 적절히 잘 구현된 거 같습니다.
